runtime/restart/monitor/monitor.go

```diff
@@ -171,9 +171,15 @@ func (m *monitor) monitor(ctx context.Context) ([]change, error) {
 		// which will result in an `on-failure` restart policy reconcile error.
 		switch desiredStatus {
 		case containerd.Running:
-			if !restart.Reconcile(status, labels) {
+			switch status.Status {
+			case "", containerd.Stopped, containerd.Unknown:
+				if !restart.Reconcile(status, labels) {
+					continue
+				}
+			default:
```

Comment:
Why not update the container's label to "paused"?

Reply:
If we choose to use the `containerd.io/restart.status=paused` label to control the restart behavior, it may complicate things a bit.

First we need to check if the label is set when we pause the container, and if it is set, then we need to update the label to pause and when resuming the container, we need to restore the label to its original value(eg, running)

This makes the behavior of the pause container and auto-restart have a relatively strong coupling

In addition the user just update the label also need to user or command to check whether you can set `running`, to prevent abnormal behavior, for example `nerdctl update --restart` logic will become much more complicated.

Just setting the label means that containers in the `Created` state may still be restarted more than once, which may cause the `on-failure:n` restart policy to become inaccurate.

Reply:
Wouldn't it be better to think of the pause as a special case of the running state?
```go
                        switch status.Status {
                        case containerd.Paused, containerd.Pausing:
                                continue
                        default:
                        }
                        if !restart.Reconcile(status, labels) {
                                continue
                        }
```

Reply:
Wouldn't it be better to think of the pause as a special case of the running state?
```go
                        switch status.Status {
                        case containerd.Paused, containerd.Pausing:
                                continue
                        default:
                        }
                        if !restart.Reconcile(status, labels) {
                                continue
                        }
```

Reply:
Rather than changing the monitor's core logic, how about adding a new policy like "unless-pause"?

Reply:
If the `unless-pause` policy is added, then it looks like we'll be using `unless-pause` more often than `always`. The inability to pause looks more like a special case.

I think whether we change the logic of always depends on whether the inability to pause is a correct phenomenon or a bug, and if we think it's a bug, then we should change the logic of monitor, and if it fits the logic of always or the default policy, then we can add a new policy(`unless-pause`), although the new policy will probably be more commonly used.

@ktock @AkihiroSuda what do you think